.azure-pipelines/client.yml

```diff
@@ -26,8 +26,8 @@ jobs:
     displayName: 'Build and Package'
     inputs:
       mavenPomFile: $(pomFile)
-      goals: 'checkstyle:check javadoc:jar source:jar package'
-      options: '--batch-mode -Dmaven.test.skip=true $(mvnFlags)'
+      goals: 'javadoc:jar source:jar package'
+      options: '--batch-mode -Dmaven.test.skip=true -Ptemplate-module $(mvnFlags)' # We include template-module so we ensure it always builds in CI
```

Comment:
What happens if you pass in this "-P" option while building a pom file that isn't the root pom and just an individual project pom like keyvault? Will maven ignore it or will it fail complaining it cannot find this profile?

Reply:
Maven will fail complaining about the missing profile. I didn't realise that this was going to be run on files other than pom.client.xml - is that the plan (or current state)?

Reply:
That is the plan as we want to be able to manually queue builds of just one library if we want to only ship one library. So we need to make this work in that context as well.

Reply:
Does maven let you condition the module list via `-D<param>` instead of using profiles? That way if we set an extra define then it would be a no opt if the pom file didn't support it.

Reply:
Based on some quick testing you should be able to set the profile activation to only trigger with a parameter like:

```
         <activation>
             <property>
                 <name>include-template</name>
             </property>
         </activation>
```
And then update the command line to add `-Dinclude-template`. That will be ignored when building a individual pom but when building the root will cause the template profile to activate.

Reply:
Thanks - I'm just getting started in the office so I'll quickly make this change, test, and then update the PR.